Python/errors.c

```diff
@@ -647,26 +647,6 @@ _PyErr_ChainStackItem(_PyErr_StackItem *exc_info)
     PyObject *typ, *val, *tb;
     _PyErr_Fetch(tstate, &typ, &val, &tb);
 
-    PyObject *typ2, *val2, *tb2;
-    typ2 = exc_info->exc_type;
-    val2 = exc_info->exc_value;
-    tb2 = exc_info->exc_traceback;
-#ifdef Py_DEBUG
-    PyObject *typ2_before = typ2;
-    PyObject *val2_before = val2;
-    PyObject *tb2_before = tb2;
-#endif
-    _PyErr_NormalizeException(tstate, &typ2, &val2, &tb2);
-#ifdef Py_DEBUG
-    /* exc_info should already be normalized */
-    assert(typ2 == typ2_before);
-    assert(val2 == val2_before);
-    assert(tb2 == tb2_before);
-#endif
-    if (tb2 != NULL) {
-        PyException_SetTraceback(val2, tb2);
-    }
-
```

Comment:
Is this deletion supposed to be part of this PR? (The PR title didn't indicate a cleanup in errors.c.)

@1st1 should probably review this.

Reply:
It is intended, I thought the error.c change doesn’t need to be in news because it’s a private function.

Yes, I’ll wait to hear from @1st1.

Once we have just exc_value in StackItem, I think we should add something like a PyErr_FetchNormalized(&exc) for these cases. Then you use PyErr_Fetch only for Fetch-Restore.

Reply:
Actually it's a good point, I'll remove this from this PR so it's just asyncio and @1st1 won't need to worry about this part.